packages/mdc-slider/foundation.ts

```diff
@@ -36,11 +36,14 @@ type MoveEventMap = {
   readonly[K in DownEventType]: MoveEventType;
 };
 
-const hasPointer = !!window.PointerEvent;
+// Accessing `window` without a `typeof` check will throw on Node environments.
+const hasWindow = typeof window !== 'undefined';
+const hasPointerEventSupport = hasWindow && !!window.PointerEvent;
```

Comment:
This looks good to me.

I think the right fix would be to add adapter method to check if window object has `PointerEvent`. Ideally foundation business logic shouldn't directly access DOM related objects.

WDYT? /cc @e111077

Reply:
I would agree. I didn't know whether there was a reason to have it set up there. i.e. not having to construct a new array on every foundation creation.